backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java

```diff
@@ -112,8 +113,24 @@ public Conversation editTitle(Conversation conversation, String title) throws Un
             throw new UnAuthorizedException();
         }
 
-        return conversationPersistenceManager.save(
-                conversation.toBuilder().title(title).build(), user.get());
+        return getConversation(conversation, title, user);
+    }
+
+    private Conversation getConversation(Conversation conversation, String title, Optional<User> user) {
+        conversation = Conversation.builder()
+                .id(conversation.getId())
+                .discussions(conversation.getDiscussions())
+                .title(getTitle(conversation, title, user))
+                .createdOn(conversation.getCreatedOn())
+                .lastUpdatedOn(Instant.now())
+                .build();
+        return conversation;
+    }
+
+    private String getTitle(Conversation conversation, String title, Optional<User> user) {
```

Comment:
this does not solve anything you are just saving in a get method that is not a good behaviour